Makefile

```diff
@@ -0,0 +1,18 @@
+.PHONY: all
+
+all: format build test
+
+build:
+	@cargo build 
+
+test:
+	@export RUST_BACKTRACE=1 && cargo test -- --nocapture 
+
+format: 
+	@cargo fmt -- --write-mode diff | grep -E "Diff .*at line" > /dev/null && cargo fmt -- --write-mode overwrite || exit 0
```

Comment:
When I change to 
```
define do-format-with-cmd
$1 --write-mode diff $2 |  grep -E "Diff .*at line" > /dev/null && $1 --write-mode overwrite $2 || exit 0
endef

format: 
	$(call format-code-in,.)
	$(call format-code-in,tests/test.rs)
	$(call format-code-in,librocksdb_sys)
```

and run `make format`, it outputs `make: `format' is up to date.` and formats nothing.